### DISCUSSION TITLE
async API for subscription data changes instead of callbacks

### DISCUSSION THREAD

**Discussion by bgusach** [FreeOpcUa/opcua-asyncio#1407] [Ideas]
Hello, 

I've recently ported a somewhat complex code base from the old `opcua`  library to `asyncua`, and while it definitely feels way better now, there's one itchy point that doesn't feel completely right: subscriptions still work with callbacks. This forces me to pass certain callbacks through a couple of APIs until I can attach it to the `SubHandler`.

I'm by no means a specialist, but I think it could be possible to `await` some coroutine method on `Subscription`. I think reducing callback complexity is one of the main reasons for the whole `asyncio` shebang. 

Would that make sense? Thanks.

**Comment by oroulet** [comment]
Yes that make sense. Now i remember that also was part of the original plan, but either there is undocumented code somewhere or we just forgot it in the meantime....
Almost like I want to have a look.... But feel free to try and post something
